pkg/csi/service/vanilla/controller_helper.go

```diff
@@ -295,3 +295,35 @@ func GetVolumeManagerFromVCHost(ctx context.Context, managers *common.Managers,
 	}
 	return volumeMgr, nil
 }
+
+// getCandidateDSInTopologyForFileVolumes returns a list of shared accessible file service
+// enabled datastores for requested topology.
+func getCandidateDSInTopologyForFileVolumes(ctx context.Context,
+	topologyArr []*csi.Topology, vcenter *vsphere.VirtualCenter) ([]*vsphere.DatastoreInfo,
+	error) {
+	log := logger.GetLogger(ctx)
+
+	var sharedDatastores []*vsphere.DatastoreInfo
+	// A topology requirement is an array of topology segments.
+	for _, topology := range topologyArr {
+		segments := topology.GetSegments()
+		hostMoRefs, err := common.GetHostsForSegment(ctx, segments, vcenter)
+		if err != nil {
+			return nil, logger.LogNewErrorCodef(log, codes.Internal,
+				"failed to fetch hosts belonging to topology %+v. Error: %+v", topologyArr, err)
+		}
+		sharedDatastores, err = vsphere.GetSharedDatastoresForHosts(ctx, hostMoRefs)
```

Comment:
sharedDatastores is getting over ridden in the `for _, topology := range topologyArr {` loop.

Reply:
Thanks for catching this. I'm appending it now